src/com/facebook/buck/jvm/java/PluginFactory.java

```diff
@@ -0,0 +1,47 @@
+package com.facebook.buck.jvm.java;
+
+import com.facebook.buck.util.ClassLoaderCache;
+import com.google.common.collect.ImmutableList;
+import java.net.URL;
+
+// Counter part of AnnotationProcessorFactory
+public class PluginFactory implements AutoCloseable {
```

Comment:
Can we add an integration test?

Reply:
I've found a test on `AnnotationProcessorFactory`. Will replicate it's behaviour then in a test for this class, since they are very similar.
But overall there is no integration test that I could find for AP as well, so I actually have no clue on how to check if classes are being reloaded or not in an integration test.
If you have any suggestion please do tell :)